external/kafka/src/main/scala/org/apache/spark/streaming/kafka/KafkaRDD.scala

```diff
@@ -106,7 +106,7 @@ class KafkaRDD[
   override def getPreferredLocations(thePart: Partition): Seq[String] = {
     val part = thePart.asInstanceOf[KafkaRDDPartition]
     // TODO is additional hostname resolution necessary here
-    Seq(part.host)
+    Seq()
```

Comment:
this was causing the ConstrainedTaskPlacer in sparrow to always pick only First Executor out of a list of Executors by IP and thus tasks go to only 1 backend. Probably a bug in ConstrainedTaskPlacer, but i worked around by removing the preferredLocation which makes the UnconstrainedTaskPlacer to handle the request